src/compile.rs

```diff
@@ -329,6 +347,158 @@ pub fn build_func_map<'a>(
     Ok(decl_map)
 }
 
+pub fn validate_fs_context_duplicates(
+    fsc_rules: BTreeMap<&String, BTreeSet<&FileSystemContextRule>>,
+) -> Result<(), CascadeErrors> {
+    let mut errors = CascadeErrors::new();
+
+    'key_loop: for (_, v) in fsc_rules {
```

Comment:
I think this is equivalent to `for v in fsc_rules.values()` and the values() version is more idiomatic.